File: 백준/Silver/3135. 라디오/라디오.py
import sys

A, B = map(int, sys.stdin.readline().strip().split())
N = int(sys.stdin.readline())
arr = []
for i in range(N):
  arr.append(int(sys.stdin.readline()))

temp = abs(A - B)

flag = True
for i in range(N):
  if temp > abs(arr[i] - B):
    flag = False
    temp = abs(arr[i] - B)

if not flag:
  print(temp+1)
else:
  print(temp)

File: 백준/Silver/3135. 라디오/라디오.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {


    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int gap = Integer.MAX_VALUE;

        String[] sa = bf.readLine().split(" ");
        int a = Integer.parseInt(sa[0]);
        int b = Integer.parseInt(sa[1]);
        int n = Integer.parseInt(bf.readLine());
        int num = Math.abs(b - a);
        int[] arr = new int[n];

        int count = 0;
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(bf.readLine());
            gap = Math.min(gap, Math.abs(b - arr[i]));
        }

        if (num <= gap) count = num;
        else count = gap + 1;

        System.out.println(count);


    }
}
